cmd/operator-sdk/generate/crds.go

```diff
@@ -54,11 +55,19 @@ func crdsFunc(cmd *cobra.Command, args []string) error {
 	if len(args) != 0 {
 		return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 	}
+	projutil.MustInProjectRoot()
 
-	// Skip usage printing on error, since this command will never fail from
-	// improper CLI usage.
-	if err := genutil.CRDGen(crdVersion); err != nil {
-		log.Fatal(err)
+	switch t := projutil.GetOperatorType(); t {
```

Comment:
Rather than switching on this, can we define a new switch that checks for the presence of `pkg/apis` and `go.mod`, as it seems that those are the only required files for generation to succeed. This will make it so that users that provide those files can still use these functions in non-Go-based operators

Reply:
Would they have `pkg/apis` if `cmd/manager/main.go` wasn't also present?

Reply:
currently they would have to create `pkg/apis` themselves (or do something hacky like `touch main.go` before running it so that it's detected as a go-type project), but if there is a `pkg/apis` present then it works whether or not the rest of the scaffolded go code is there

Reply:
I'm not sure if there are any users out there who have figured out that this works, but if there are it doesn't seem to be technically necessary for us to block it off. We don't need to explicitly support it (and if we want to should probably add options to the Ansible scaffolding to generate these files), but if a user wants to generate types in this way I see no reason to add additional barriers.

Reply:
IMO since we do not support the gen k8s and cRD this shows fine. 

However, if we would like to follow up @fabianvf suggestion we can just call the `genutil.CRDGen` for Ansible and Helm and print an alert informing that is not supported instead of the block as well.